### ISSUE TITLE
implosion compressor giving false output

### ISSUE THREAD

**Issue by Mooshimi** [tgstation/tgstation#61420]
Reporting client version: 514.1561

## Round ID:
Irrelevant - Happens most the time I've seen

## Testmerges:
- [Radically improves explosion logging.](https://www.github.com/tgstation/tgstation/pull/61419)
- [Sparring Sect!](https://www.github.com/tgstation/tgstation/pull/61316)

## Reproduction:
Us the toxins explosion compressor to refine a core. It shows the error message and also the success message, and outputs a successful core. 
![image](https://user-images.githubusercontent.com/85910816/132971733-fce5e73f-6893-4df4-acc5-a2b515e5f3d0.png)

**Comment by TemporalOroboros** [comment]
You're clicking on the explosion compressor after the core has been compressed and it's trying to find the explosion radius required to refine the refined core.

**Comment by Mooshimi** [comment]
> You're clicking on the explosion compressor after the core has been compressed and it's trying to find the explosion radius required to refine the refined core.

But if this is the first success message and it comes after that, how does that make sense? Doesn't it just spit the core out after it compresses anyways?

**Comment by TemporalOroboros** [comment]
> 
> 
> > You're clicking on the explosion compressor after the core has been compressed and it's trying to find the explosion radius required to refine the refined core.
> 
> But if this is the first success message and it comes after that, how does that make sense? Doesn't it just spit the core out after it compresses anyways?

Nope. It actually returns the items after a set delay. We can't actually guarantee that the TTV you stick into it is going to explode. The user might have jammed two tanks directly from the tank holder onto the transfer valve and called it a day. As such, if we dispense them when the TTV explodes we might wind up locking the machine forever when it doesn't.

**Comment by Mooshimi** [comment]
> > > You're clicking on the explosion compressor after the core has been compressed and it's trying to find the explosion radius required to refine the refined core.
> > 
> > 
> > But if this is the first success message and it comes after that, how does that make sense? Doesn't it just spit the core out after it compresses anyways?
> 
> Nope. It actually returns the items after a set delay. We can't actually guarantee that the TTV you stick into it is going to explode. The user might have jammed two tanks directly from the tank holder onto the transfer valve and called it a day. As such, if we dispense them when the TTV explodes we might wind up locking the machine forever when it doesn't.
> 
> If we based retrieving the items off of clicking on the machine that would probably be better... But my mouse specifically has an issue with spontaneously double-clicking. I don't exactly want to try doing toxins only to eject a primed maxcap directly into my face.

Should it give the success line earlier then? So they don't get confused? I see a lot of people confused by it. And maybe a message tacked on about dispensing the core in a few seconds?
